internal/auditlog/serial_writer.go

```diff
@@ -0,0 +1,52 @@
+// Copyright The OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package auditlog
+
+import (
+	"io"
+
+	"github.com/corazawaf/coraza/v3/experimental/plugins"
+	"github.com/corazawaf/coraza/v3/experimental/plugins/plugintypes"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm"
+)
+
+// Coraza does not come with a built-in audit log writer for Wasm
+// See https://github.com/corazawaf/coraza/blob/main/internal/auditlog/init_tinygo.go
+// This function registers a new audit log writer for Wasm named "wasmserial" that prints
+// audit logs to the proxy-wasm log as info messages.
+func RegisterWasmSerialWriter() {
+	plugins.RegisterAuditLogWriter("wasmserial", func() plugintypes.AuditLogWriter {
```

Comment:
I would avoid conflating the writer with technology. Can we override the serial in here only?